### ISSUE TITLE
Inconsistent behavior in assign+bindsym?

### ISSUE THREAD

**Issue by chronokitsune3233** [swaywm/sway#1168]
Config file:

```
set $mod Mod4
bindsym $mod+Return lxterminal
bindsym $mod+1 workspace 1
bindsym $mod+2 workspace 2
assign [class="Lxterminal"] 1
```

Problem:

Launching my terminal emulator using the key combination specified in the bindsym works, except the workspace isn't switched. Launching a second instance of my terminal emulator via the same key combination will switch workspaces as expected.

Steps to reproduce the problem:

1. Execute sway from tty.
2. Switch to workspace 2 using $mod+2.
3. Press $mod+Return to create the first instance of lxterminal (or your terminal emulator of choice).
4. The instance of lxterminal is created on workspace 1.
5. Workspace 2 is still the current workspace; the focus was not switched to lxterminal on workspace 1.
6. Press $mod+Return to create a second instance of lxterminal.
7. The workspace switches to workspace 1 and the new instance of lxterminal is focused.

Executing lxterminal works just fine via dmenu:

1. Execute sway from tty.
2. Switch to workspace 2 using $mod+2.
3. Use dmenu to execute lxterminal.
4. The workspace switches to workspace 1 and the new instance of lxterminal is focused.

My workaround is the following:

```
for_window [class="Lxterminal"] exec swaymsg 'move container to workspace 1' && swaymsg 'workspace 1'
```

Should the workspace be switched every time a window matching the criteria appears&mdash;whether it was launched via bindsym or not, similar to my workaround&mdash;or is this behavior intentional? Not limited to lxterminal, this behavior also occurs with Chromium, weston-terminal, and any other application; the Arch sway package and my own build compiled from source both exhibit the same behavior.

```
$ sway --version
sway version 0.12-rc1-69-gcd3c48f (2017-04-11, branch "master")
```

**Comment by SirCmpwn** [comment]
Does i3 switch the workspace to the one the new window is created on? I didn't think it did.

**Comment by chronokitsune3233** [comment]
It doesn't. i3 doesn't switch the workspace at all; I assumed sway's behavior was an improvement over i3's behavior.

**Comment by SirCmpwn** [comment]
No, that's not supposed to happen. The bug here is that it automatically switches to 1 the second time.

**Comment by DmitryHetman** [comment]
Does LXterminal support wayland?

**Comment by chronokitsune3233** [comment]
@DmitryHetman When built with GTK+ 3.x libraries, yes. Otherwise, no. The Arch package lxterminal-gtk3 provides such a build.

However, you question did prompt me to try the same thing with Wayland Terminal (`weston-terminal`). It _did not_ switch to workspace 1 the second time as it does with LXTerminal.

`gtk3-demo-application`, Chromium, ROXTerm, LXTerminal, and Xfce Terminal all use the GTK+ 3.0 libs and exhibit the same problem. Strangely, the Sakura terminal emulator, the Termite terminal emulator, and `gtk3-demo` do not suffer from the issue.

I've attached a unified diff of the differences in the debug info I collected for the ROXTerm and Sakura terminal emulators; for simpler comparison, I changed the PIDs and output/workspace/container/view addresses.

The interesting stuff happens at line 83 when the second identical process is started for each respective session (again, via bindsym), then later again at line 133 when the ROXTerm session destroys the original workspace (as if I had manually switched from an empty workspace to the ROXTerm workspace) while the Sakura session simply displays the original "Web" workspace, switching back after the window is mapped(?)

I'm honestly still not certain what is causing this, but in both instances, `assign`/`for_window` matches the application fine as evidenced by the multiple windows in the other workspace, yet ROXTerm and friends inexplicably force a workspace switch. I've attached the config file I've been using to test this.

[logdiff_u.txt](https://github.com/SirCmpwn/sway/files/924008/logdiff_u.txt)
[config.txt](https://github.com/SirCmpwn/sway/files/924022/config.txt)

**Comment by chronokitsune3233** [comment]
It would seem that when an application manages its own windows, including when you start a new process, there is a switch to the new window, yet i3 somehow prevents this from happening. Then again, maybe there's a window hint or notification of some type available for X that isn't available with Wayland yet, so the focus is stolen by the second new window in Sway?

**Comment by SirCmpwn** [comment]
I'm closing issues that are resolved by the yet-unreleased sway 1.0.
